source/code/plugins/oms_common.rb

```diff
@@ -478,6 +480,141 @@ class Common
     @@tzRightFolder = 'right/'
 
     class << self
+
+      # Internal methods
+      # (left public for easy testing, though protected may be better later)
+
+      def clean_hostname_string(hnBuffer)
+        s = hnBuffer.strip
+        # Take off any port numbers:
+        if /\A(\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}):\d+\z/ =~ s
+            return $1
+        elsif not is_like_ipv6_string?(hnBuffer) and /\A(.*):\d+\z/ =~ s
+            return $1
+        end
+        return s
+      end
+
+      def has_Designated_HostnameFile?
+        return false if @@HostnameFilePath.nil?
+        return false if @@HostnameFilePath.length < 8 # 8 is first Guess.  Need policy here.
+        File.exist?(@@HostnameFilePath)
+      end
+
+      def is_dot_separated_string?(hnBuffer)
+        return true if /[^.]+\.[^.]+/ =~ hnBuffer
+        return false
+      end
+
+      def is_hostname_compliant?(hnBuffer)
+        # RFC 2181:
+        #   Size limit is 1 to 63 octets, so probably bytesize is appropriate method.
+        return false if hnBuffer.nil?
+        return false if /\./ =~ hnBuffer # Hostname by definition may not contain a dot.
+        return false if /:/ =~ hnBuffer # Hostname by definition may not contain a colon.
+        return true if 1 <= hnBuffer.bytesize && hnBuffer.bytesize <= 63
+        return false
+      end
+
+      def is_like_ipv4_string?(hnBuffer)
+        return false unless /\A[12]?\d{1,2}\.[12]?\d{1,2}\.[12]?\d{1,2}\.[12]?\d{1,2}\z/ =~ hnBuffer
```

Comment:
For reference, where did you find this regex? Are we sure it won't exclude valid cases?

Reply:
If someday the "Computer/Hostname" fields are used for IPs of network devices forwarding data, this may need to be more broad. Simply validating that it has four parts of all digits should be our bar because of strange cases like this.

Reply:
That is something I came up with.  It is not all encompassing.  It will include many things that are outside the proper range, and now that I have the loop below it checking the octets, it is probably better to just to back to \d+ and just use the loop to check the ranges.  Unless you have other suggestions that's what I'll do.